kubernetes/client/models/v1_node_condition.py

```diff
@@ -214,12 +214,13 @@ def type(self, type):
         """
         if self.local_vars_configuration.client_side_validation and type is None:  # noqa: E501
             raise ValueError("Invalid value for `type`, must not be `None`")  # noqa: E501
-        allowed_values = ["DiskPressure", "MemoryPressure", "NetworkUnavailable", "PIDPressure", "Ready"]  # noqa: E501
+        allowed_values = ["DiskPressure", "MemoryPressure", "NetworkUnavailable", "PIDPressure", "Ready", "TerminateScheduled", "PreemptScheduled", "FreezeScheduled", "RebootScheduled", "FrequentContainerdRestart", "RedeployScheduled", "ReadonlyFilesystem", "FrequentKubeletRestart", "ContainerRuntimeProblem", "FilesystemCorruptionProblem", "FrequentDockerRestart", "KubeletProblem", "KernelDeadlock","FrequentUnregisterNetDevice"]  # noqa: E501
```

Comment:
I think this needs to be fixed in the upstream kubernetes. This client is generated from Kubernetes OpenAPI spec, and according to the [spec](https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/swagger.json):
```
        "type": {                                                                                                                       
          "description": "Type of node condition.\n\nPossible enum values:\n - `\"DiskPressure\"` means the kubelet is under pressure due to insufficient available disk.\n - `\"MemoryPressure\"` means the kubelet is under pressure due to insufficient available memory.\n - `\"NetworkUnavailable\"` means that network for the node is not correctly configured.\n - `\"PIDPressure\"` means the kubelet is under pressure due to insufficient available PID.\n - `\"Ready\"` means kubelet is healthy and ready to accept pods.",                        
          "enum": [                                                                                                                                 
            "DiskPressure",                                                                                                             
            "MemoryPressure",                                                                                                           
            "NetworkUnavailable",                                                                                                       
            "PIDPressure",                                                                                                              
            "Ready"                                                                                                                               
],                                                                                                                                      
"type": "string"                                                                                                                      
}
```

Reply:
@roycaihw you reckon we should make a PR over there? any guidelines on where should we add this ?

Reply:
Yes, I'm afraid we should remove [this line](https://github.com/kubernetes/kubernetes/blob/5211a4b2147ce745649e77687a87cdebc08a6707/staging/src/k8s.io/api/core/v1/types.go#L5053). 

https://github.com/kubernetes/kubernetes/pull/105057 made the openapi-generated clients do more strict validation for this enum. However, according to https://github.com/kubernetes-client/python/issues/1735 it seems common for cloud providers (e.g. GKE, AKS) to expend the possible values for this enum-- which won't pass the client-side validation.

Reply:
In the meantime, I think we can accept a [hot patch](https://github.com/kubernetes-client/python/blob/master/scripts/apply-hotfixes.sh) to remove this client-side validation. It seems pretty bad that the client will randomly fail listing nodes when talking to a cloud provider.

Reply:
I think we should treat this field to be not an enum. I will send a PR shortly.